source/ThemeManager.cpp

```diff
@@ -0,0 +1,145 @@
+        CFPreferencesAppSynchronize(kCFPreferencesCurrentApplication);
+
+        // Try to get the interface style from user defaults
+        CFPropertyListRef propertyList = CFPreferencesCopyAppValue(
+            CFSTR("AppleInterfaceStyle"),
+            kCFPreferencesCurrentApplication
+        );
+
+        if (propertyList && CFGetTypeID(propertyList) == CFStringGetTypeID()) {
+            appearanceName = (CFStringRef)propertyList;
+            if (CFStringCompare(appearanceName, CFSTR("Dark"), 0) == kCFCompareEqualTo) {
+                CFRelease(propertyList);
+                return "dark";
+            }
+        }
+
+        if (propertyList) {
+            CFRelease(propertyList);
+        }
+        return "light";
+
+#elif __linux__
+        // Try GNOME gsettings
+        const char *gsettings_cmd = "gsettings get org.gnome.desktop.interface gtk-theme 2>/dev/null";
+        FILE *pipe = popen(gsettings_cmd, "r");
+        if (pipe) {
+            char buffer[256];
+            if (fgets(buffer, sizeof(buffer), pipe)) {
+                std::string theme(buffer);
+                pclose(pipe);
+
+                std::transform(theme.begin(), theme.end(), theme.begin(), ::tolower);
+                if (theme.find("dark") != std::string::npos) {
+                    return "dark";
+                }
+            } else {
+                pclose(pipe);
+            }
+        }
+
+        // Try KDE plasma settings
+        const char *kde_cmd = "kreadconfig5 --group General --key ColorScheme 2>/dev/null";
+        pipe = popen(kde_cmd, "r");
+        if (pipe) {
+            char buffer[256];
+            if (fgets(buffer, sizeof(buffer), pipe)) {
+                std::string scheme(buffer);
+                pclose(pipe);
+                std::transform(scheme.begin(), scheme.end(), scheme.begin(), ::tolower);
+                if (scheme.find("dark") != std::string::npos || scheme.find("breeze dark") != std::string::npos) {
+                    return "dark";
+                }
+            } else {
+                pclose(pipe);
+            }
+        }
+
+        // Check environment variables
+        const char *gtk_theme = std::getenv("GTK_THEME");
+        if (gtk_theme) {
+            std::string theme(gtk_theme);
+            std::transform(theme.begin(), theme.end(), theme.begin(), ::tolower);
+            if (theme.find("dark") != std::string::npos) {
+                return "dark";
+            }
+        }
+
+        return "light";
+
+#else
+        // Unsupported platform - fallback to dark
+        return "dark";
+#endif
+    }
+
+    // TODO: remove after we store the value inside SQLite
+    void ThemeManager::initializeTheme() {
+        std::string actualTheme = this->currentTheme;
+
+        if (this->currentTheme == "system") {
+            actualTheme = detectSystemTheme();
+        }
+
+        this->currentTheme = actualTheme;
+        tgui::Theme::setDefault("assets/themes/" + actualTheme + ".txt");
```

Comment:
Redundant logic with `setTheme()`
```suggestion
        ThemeManager::setTheme(this->currentTheme);
```